omnibus/files/private-chef-scripts/gather-logs

```diff
@@ -176,9 +177,9 @@ then
     found_manage=true
 fi
 
-if [[ -e /opt/opscode-reporting/bin/opscode-reporting-ctl ]];
+#Use socket to make sure we're on the primary in an HA system
+if [[ -e /opt/opscode-reporting/bin/opscode-reporting-ctl && -S "/tmp/.s.PGSQL.5432" ]];
```

Comment:
So with the support we have in place, chef-server-ctl psql opscode- reporting (or maybe opscode_reporting) --as-admin should  connect if it's available, and otherwise not.  The drawback is that it won't work if reporting is configured with its own external DB, independent of chef server. That's not a concern now as the version that supports this was only just released yesterday, and it's not explicitly called out as a thing that you can do in any case - but at some point down the road, it may become one.